### PR TITLE
Fix nasty 500 error on POST to /_config

### PR DESCRIPTION
**Pull request by wenli200133**
## Overview

When the ini config file is read only,  ini file shouldn't be updated when the endpoint of `_node/_local/_config/` is invoked. 

For example:
```
curl -X PUT localhost:15984/_node/_local/_config/admins/joan -d '"500"'
```
when you check the local.ini file, you shouldn't find out the `joan` key and hash value of `500`

and it also should not return this error like below:
```
{"error":"badmatch","reason":"{badrpc,{'EXIT',{{{badmatch,{error,eacces}},\n  
```
expected returned error should like this:

```
{"error":"bad_request","reason":"eacces"}
```
## Testing recommendations

```
wenwl@wenwl_mbp:~/cloudant/couchdb$ chmod a-w ./dev/lib/node1/etc/local.ini
wenwl@wenwl_mbp:~/cloudant/couchdb$ curl -X PUT localhost:15984/_node/_local/_config/admins/joan1 -d '"500"' -u adm:pass
{"error":"bad_request","reason":"eacces"}
wenwl@wenwl_mbp:~/cloudant/couchdb$ curl -X DELETE localhost:15984/_node/_local/_config/admins/joan1 -u adm:pass
{"error":"bad_request","reason":"eacces"}
```

## Related Issues or Pull Requests

https://github.com/apache/couchdb/issues/1380
https://github.com/apache/couchdb-config/pull/21

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;
- [ ] Documentation reflects the changes;

### CODE REVIEW

**Comment by wohali** [comment]
Your tests are failing.

Does this impact what happens if I do something like:

```
curl -X PUT localhost:15984/_node/_local/_config/admins/joan -d '500'
```

i.e., an invalid body?

**Comment by wenli200133** [comment]
Hi @wohali , now checks have passed.
I think `'500'` is invalid body, `"500"` is valid body.
I have another PR for fixing this issue, https://github.com/apache/couchdb/pull/1717 and https://github.com/apache/couchdb-config/pull/21,  https://github.com/apache/couchdb/pull/1717 is dependent on https://github.com/apache/couchdb-config/pull/21.

**Comment by wenli200133** [comment]
PR apache/couchdb-config#21 need first enter into the master, then #1717 can pick the updated config codes up.

**Comment by wenli200133** [comment]
Hi @wohali can you help review this PR again if you have free time? @jiangphcn can you also help with it?
Thanks.

File: src/chttpd/src/chttpd_misc.erl [hunk 1]
```diff
@@ -323,8 +323,12 @@ handle_node_req(#httpd{method='DELETE',path_parts=[_, Node, <<"_config">>, Secti
     undefined ->
         throw({not_found, unknown_config_value});
     OldValue ->
-        call_node(Node, config, delete, [Section, Key, Persist]),
-        send_json(Req, 200, list_to_binary(OldValue))
+        case call_node(Node, config, delete, [Section, Key, Persist]) of
+            ok ->
+               send_json(Req, 200, list_to_binary(OldValue));
+            {error, Reason} ->
+                chttpd:send_error(Req, {bad_request, Reason})
```
Comment (jiangphcn) [inline src/chttpd/src/chttpd_misc.erl]: space

Comment (wenli200133) [reply src/chttpd/src/chttpd_misc.erl]: Done

**Comment by jiangphcn** [comment]
+1

**Pull request merged**